src/dune_rules/install_rules.ml

```diff
@@ -202,18 +202,21 @@ end = struct
         let set_dir m =
           List.map ~f:(fun (cm_kind, p) -> (cm_dir m cm_kind, p))
         in
-        let modules_impl =
-          List.concat_map installable_modules.impl ~f:(fun m ->
-              let cmt_files =
-                List.concat_map Ml_kind.all ~f:(fun ml_kind ->
-                    let open Lib_mode.Cm_kind in
-                    List.concat_map
-                      [ (native || byte, Ocaml Cmi); (melange, Melange Cmi) ]
-                      ~f:(fun (condition, kind) ->
-                        if_ condition
-                          ( kind
-                          , Obj_dir.Module.cmt_file obj_dir m ~ml_kind
-                              ~cm_kind:kind )))
+        let+ modules_impl =
+          Memo.List.concat_map installable_modules.impl ~f:(fun m ->
+              let+ cmt_files =
+                let+ bin_annot = Super_context.bin_annot sctx ~dir in
```

Comment:
This can be evaluated once to see if we need to iterate through the list of modules.

Reply:
Fixed, thanks.